[Skia] Add Color operators to easily convert from/to SkColor and SkColor4f

carlosgcampos opened the pull request:
#### fc13b190064d364d3046e56370ce5202826655fc
<pre>
[Skia] Add Color operators to easily convert from/to SkColor and SkColor4f
<a href="https://bugs.webkit.org/show_bug.cgi?id=269978">https://bugs.webkit.org/show_bug.cgi?id=269978</a>

Reviewed by Nikolas Zimmermann.

We avoid duplicated code and simplify the use of colors.

* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/skia/ColorSkia.cpp: Added.
(WebCore::Color::Color):
(WebCore::Color::operator SkColor const):
(WebCore::Color::operator SkColor4f const):
* Source/WebCore/platform/graphics/skia/GradientSkia.cpp:
(WebCore::Gradient::shader):
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
(WebCore::GraphicsContextSkia::createDropShadowFilterIfNeeded const):
(WebCore::GraphicsContextSkia::createFillPaint const):
(WebCore::GraphicsContextSkia::createStrokePaint const):

Canonical link: <a href="https://commits.webkit.org/275308@main">https://commits.webkit.org/275308@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d660103efe76d0792de06d10a8ba9255ff3e867

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41258 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20272 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43636 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43822 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37350 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43565 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23376 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17602 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34120 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41832 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17230 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35539 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14777 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14960 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36541 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45148 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37477 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36859 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40597 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16073 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13181 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38983 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17692 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17744 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5542 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17336 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

GeorgesStavracas reviewed (commented):
LGTM, looks like a nice cleanup :slightly_smiling_face:

carlosgcampos committed:
[Skia] Add Color operators to easily convert from/to SkColor and SkColor4f
https://bugs.webkit.org/show_bug.cgi?id=269978

Reviewed by Nikolas Zimmermann.

We avoid duplicated code and simplify the use of colors.

* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/skia/ColorSkia.cpp: Added.
(WebCore::Color::Color):
(WebCore::Color::operator SkColor const):
(WebCore::Color::operator SkColor4f const):
* Source/WebCore/platform/graphics/skia/GradientSkia.cpp:
(WebCore::Gradient::shader):
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
(WebCore::GraphicsContextSkia::createDropShadowFilterIfNeeded const):
(WebCore::GraphicsContextSkia::createFillPaint const):
(WebCore::GraphicsContextSkia::createStrokePaint const):

Canonical link: https://commits.webkit.org/275308@main
